engine/utils/file_manager_utils.h

```diff
@@ -84,23 +85,54 @@ inline std::filesystem::path GetConfigurationPath() {
   return configuration_path;
 }
 
-inline std::filesystem::path GetCortexPath() {
+inline void CreateConfigFileIfNotExist() {
+  auto config_path = file_manager_utils::GetConfigurationPath();
+  if (std::filesystem::exists(config_path)) {
+    // already exists
+    return;
+  }
+  CLI_LOG("Config file not found. Creating one at " + config_path.string());
+  auto defaultDataFolderPath =
+      file_manager_utils::GetHomeDirectoryPath() / config_yaml_utils::kCortexFolderName;
+  auto config = config_yaml_utils::CortexConfig{
+      .dataFolderPath = defaultDataFolderPath.string(),
+      .host = config_yaml_utils::kDefaultHost,
+      .port = config_yaml_utils::kDefaultPort,
+  };
+  std::cout << "config: " << config.dataFolderPath << "\n";
+  DumpYamlConfig(config, config_path.string());
+}
+
+inline config_yaml_utils::CortexConfig GetCortexConfig() {
+  auto config_path = GetConfigurationPath();
+  std::string variant = "";  // TODO: empty for now
+  return config_yaml_utils::FromYaml(config_path.string(), variant);
+}
+
+inline std::filesystem::path GetCortexDataPath() {
   // TODO: We will need to support user to move the data folder to other place.
   // TODO: get the variant of cortex. As discussed, we will have: prod, beta, nightly
-  // currently we will store cortex data at ~/.cortex
+  // currently we will store cortex data at ~/.cortexcpp
+  auto config = GetCortexConfig();
+  std::filesystem::path data_folder_path;
+  if (!config.dataFolderPath.empty()) {
+    data_folder_path =
+        std::filesystem::path(config.dataFolderPath) / ".cortexcpp";
```

Comment:
since default `dataFolderPath` is having `kCortexFolderName` already, do we need to add another `.cortexcpp` under it?